addOns/custompayloads/src/test/java/org/zaproxy/zap/extension/custompayloads/CustomPayloadsParamUnitTest.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2024 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.custompayloads;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.configuration.HierarchicalConfiguration;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.zaproxy.zap.utils.ZapXmlConfiguration;
+
+/** Unit test for {@link CustomPayloadsParam}. */
+class CustomPayloadsParamUnitTest {
+
+    private CustomPayloadsParam param;
+    private ZapXmlConfiguration configuration;
+
+    @BeforeEach
+    void setUp() {
+        param = new CustomPayloadsParam();
+        configuration = new ZapXmlConfiguration();
+    }
+
+    @Test
+    void shouldHaveConfigVersionKey() {
+        // Given / When
+        param.load(configuration);
+        // Then
+        assertThat(param.getConfigVersionKey(), is(equalTo("custompayloads[@version]")));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"true", "false"})
+    void shouldLoadConfirmRemoveFromConfig(boolean state) {
+        // Given
+        configuration.setProperty(CustomPayloadsParam.CONFIRM_REMOVE_PAYLOAD_KEY, state);
+        // When
+        param.load(configuration);
+        // Then
+        assertThat(param.isConfirmRemoveToken(), is(equalTo(state)));
+    }
+
+    @Test
+    void shouldDefaultConfirmRemoveTrue() {
+        // Given / When
+        param.load(configuration);
+        // Then
+        assertThat(param.isConfirmRemoveToken(), is(equalTo(true)));
+    }
+
+    @Test
+    void shouldHaveExpectedVersionInDefaultConfig() {
+        // Given / When
+        param.load(configuration);
+        // Then
+        assertThat(param.getCurrentVersion(), is(equalTo(1)));
+    }
+
+    @Test
+    void shouldNotHaveNextPayloadIdOnUpdateFromUnversioned() {
+        // Given / When
+        String nextId = configuration.getString("custompayloads.nextPayloadId");
```

Comment:
It should set this key to the configuration and then load and verify it's no longer present.